### PR TITLE
Fix TrackPendingOperation

### PR DESCRIPTION
**Pull request by EricSten-MSFT**

### CODE REVIEW

File: Kudu.Core/Helpers/PostDeploymentHelper.cs [hunk 1]
```diff
@@ -433,7 +433,7 @@ public static void RunPostDeploymentScripts(TraceListener tracer)
 
                 var cancelation = new CancellationTokenSource();
                 var delay = Task.Delay(DefaultUpdateMarkerIntervalMS, cancelation.Token);
-                var completed = await Task.WhenAny(delay, task);
+                var completed = await Task.WhenAny( new Task[] { task, delay } );
```
Comment (davidebbo) [inline Kudu.Core/Helpers/PostDeploymentHelper.cs]: Isn't that equivalent? What's the rationale behind the change?

Comment (EricSten-MSFT) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: It's not, apparently.  WhenAny has overrides, and the (delay, task) was matching one of the overrides of (Task[], Int32).  Forcing to an explicit array fixes things.

Comment (davidebbo) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: I don't understand how that's possible, since that override takes an int and we are not passing any. We also use that same overload that takes 2 tasks in other places.

Maybe something else is at play and that is not really the issue you're after?

Comment (EricSten-MSFT) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: Yes, you're right.  Please reject this PR.  I'm debugging it, and it only works under the debugger.
